examples/alert-absence-dedup/README.md

```diff
@@ -7,91 +7,119 @@ them is missing. If one or two timeseries are missing, you want exactly one
 alert. When there is a total outage you still want to get one alert, not 100
 alerts.
 
+The test app generates time series with a custom metric called
+task_latency_distribution with tags based on a partition label. Each
+partition generates its own time series.
+
 The example assumes that you are familiar with Stackdriver Monitoring and
 Alerting. It builds on the discussion in
 [Alerting policies in depth](https://cloud.google.com/monitoring/alerts/concepts-indepth).
 
-## Setup
+## Deploy the app
+
+The example code is based on the Go code in
+[Custom metrics with OpenCensus](https://cloud.google.com/monitoring/custom-metrics/open-census).
+
+[Download](https://golang.org/dl/) and install the latest version of Go.
 
-Create three GCE instances
+Build the test app
 
 ```shell
-ZONE=us-central1-c
-gcloud compute instances create test-instance-1 test-instance-2 test-instance-3 \
-  --zone=$ZONE \
-  --scopes=https://www.googleapis.com/auth/cloud-platform \
-  --boot-disk-size=20GB \
-  --machine-type=f1-micro
+go build
 ```
 
-Edit the file notification_channel.json, replacing the displayName and
-email_address. Create a notification channel with the command
+The instructions here as based on
+[Setting up authentication](https://cloud.google.com/monitoring/docs/reference/libraries#setting_up_authentication)
+for the Stackdriver client library. Note that if you run the test app on a
+Google Cloud Compute Engine instance, Google Kubernetes Engine, or App Engine
+you will not need to create a service account or download the credentials.
+
+First, set the project id in a shell variable
 
 ```shell
-gcloud alpha monitoring channels create \
-  --channel-content-from-file=notification_channel.json
+export GOOGLE_CLOUD_PROJECT=[your project]
```

Comment:
Is there a reason that you use this approach vs, say:

    gcloud config configurations create YOUR_CONFIG
    gcloud config set project PROJECT_ID

And then, elsewhere:

     $(gcloud config get-value project)

Note that the advantage of "gcloud config set project" is that most gcloud commands will then allow you to omit a project parameter entirely (with that project being used implicitly).

It's also worth noting that the project ID can also be picked up ambiently in certain situations; for example, when doing this exercise on Cloud Shell, the $DEVSHELL_PROJECT_ID variable already has the project ID.

Reply:
The environment variable GOOGLE_CLOUD_PROJECT is used by the client library in the program alert-absence-demo. In the example code here:

https://cloud.google.com/monitoring/docs/reference/libraries#setting_up_authentication

the project id is set explicitly in the code:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/monitoring/monitoring_quickstart/main.go#L43

However, the client library can also use the environment variable GOOGLE_CLOUD_PROJECT, which is more flexible. This is documented here:
https://google-auth.readthedocs.io/en/latest/reference/google.auth.environment_vars.html
https://googleapis.dev/python/google-cloud-core/latest/config.html